### DISCUSSION TITLE
Автозапуск bat с правами админа (Autostart bat file with Admin)

### DISCUSSION THREAD

**Discussion by First-Colvor** [Flowseal/zapret-discord-youtube#373] [Ideas]
Просьба добавить в начало батников код который будет запускать автоматически bat с правами админа:
Так же не рекомендую использовать двойное двоеточие для комментирования.
`
@echo off
cd /D "%~dp0"
cls
title Перезапуск bat с правами админа
rem Исправленно когда временная папка с vbs не удалялась после отработки кода, переосмыслен принцип работы

rem В КОДЕ ПРОВЕРКИ АДМИНА НЕ ДОПУСКАЕТСЯ ИСПОЛЬЗОВАНИЯ КОММЕНАРИЕВ С ДВУМЯ ДВОЕТОЧИЯМИ ":", ИСПОЛЬЗОВАТЬ ТОЛЬКО "rem"

rem Проверка наличия прав Администратора (во избежания перезапуска bat если уже есть права Админа)
WHOAMI /PRIV | find /i "SeRemoteShutdownPrivilege" > nul
if %ERRORLEVEL% == 0 goto admin

rem для работы с папкой appdata права админа не требуются

	set script_patch=%appdata%\Colvor's_code\temp

md %script_patch%

rem Создаём vbs скрипт для выдачи прав Администратора bat файлу
ver |>NUL find /v "5." && if "%~1"=="" (
  Echo CreateObject^("Shell.Application"^).ShellExecute WScript.Arguments^(0^),"1","","runas",1 >"%script_patch%\Elevating.vbs"
rem Ранее использовалось с "& goto :eof", подробнее в call /?
rem cscript.exe //nologo "%script_patch%\Elevating.vbs" "%~f0"& goto :eof
  
  cscript.exe //nologo "%script_patch%\Elevating.vbs" "%~f0"
rem del /Q "%script_patch%\Elevating.vbs"
rem timeout /t 20
  rmdir /S /Q "%script_patch%"
  exit
)

:admin

rem		 |
rem 	 |
rem		 |
rem		 |
rem		\|/
rem Ваш код
`

**Comment by Flowseal** [comment]
А теперь представим что человек сидит с нашаменным net'ом и не может нормально использовать права администратора

Нет нужды в этом
